src/stdlib_ascii.f90

```diff
@@ -179,34 +176,94 @@ pure logical function is_blank(c)
 
     !> Returns the corresponding lowercase letter, if `c` is an uppercase
     !  ASCII character, otherwise `c` itself.
-    pure function to_lower(c) result(t)
+    pure function char_to_lower(c) result(t)
         character(len=1), intent(in) :: c !! A character.
         character(len=1)             :: t
         integer :: k
 
-        k = index( upper_case, c )
+        k = index( uppercase, c )
 
         if ( k > 0 ) then
-            t = lower_case(k:k)
+            t = lowercase(k:k)
         else
             t = c
         endif
-    end function
+    end function char_to_lower
 
     !> Returns the corresponding uppercase letter, if `c` is a lowercase
     !  ASCII character, otherwise `c` itself.
-    pure function to_upper(c) result(t)
+    pure function char_to_upper(c) result(t)
         character(len=1), intent(in) :: c !! A character.
         character(len=1)             :: t
         integer :: k
 
-        k = index( lower_case, c )
+        k = index( lowercase, c )
 
         if ( k > 0 ) then
-            t = upper_case(k:k)
+            t = uppercase(k:k)
         else
             t = c
         endif
-    end function
+    end function char_to_upper
+
+    !> Convert character variable to lower case
+    pure function to_lower(string) result(lower_string)
+        character(len=*), intent(in) :: string
```

Comment:
```suggestion
        !! version: experimental
        !!
        !! Converts input character variable to all lowercase.

        character(len=*), intent(in) :: string
```

Reply:
I'll use predoc comments here to be consistent with the rest of the module.